From 911ac29b9eed3c66bba60eda6af8e5de62607e5c Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@altlinux.org>
Date: Wed, 4 Nov 2020 08:00:00 +0000
Subject: [PATCH] config: do not define DEBUGINFOD_URLS environment variable
 unnecessarily

Before this change, when elfutils was configured without
--enable-debuginfod-urls, the installed profile.d/debuginfod.sh and
profile.d/debuginfod.csh scripts used to define the DEBUGINFOD_URLS
environment variable as an non-empty string containing spaces, making
all libdebuginfod users do extra work.

Change these scripts to avoid defining the DEBUGINFOD_URLS environment
variable unless configured using --enable-debuginfod-urls.

Signed-off-by: Dmitry V. Levin <ldv@altlinux.org>
---
 config/ChangeLog      |  5 +++++
 config/profile.csh.in | 13 +++++++++++--
 config/profile.sh.in  |  7 ++++---
 3 files changed, 20 insertions(+), 5 deletions(-)

diff --git a/config/profile.csh.in b/config/profile.csh.in
index 4f25896d..0a2d6d16 100644
--- a/config/profile.csh.in
+++ b/config/profile.csh.in
@@ -1,2 +1,11 @@
-
-setenv DEBUGINFOD_URLS "$DEBUGINFOD_URLS @DEBUGINFOD_URLS@"
+if ("@DEBUGINFOD_URLS@" != "") then
+	if ($?DEBUGINFOD_URLS) then
+		if ($%DEBUGINFOD_URLS) then
+			setenv DEBUGINFOD_URLS "$DEBUGINFOD_URLS @DEBUGINFOD_URLS@"
+		else
+			setenv DEBUGINFOD_URLS "@DEBUGINFOD_URLS@"
+		endif
+	else
+		setenv DEBUGINFOD_URLS "@DEBUGINFOD_URLS@"
+	endif
+endif
diff --git a/config/profile.sh.in b/config/profile.sh.in
index 8a022489..aa228a0d 100644
--- a/config/profile.sh.in
+++ b/config/profile.sh.in
@@ -1,3 +1,4 @@
-
-DEBUGINFOD_URLS="$DEBUGINFOD_URLS @DEBUGINFOD_URLS@"
-export DEBUGINFOD_URLS
+if [ -n "@DEBUGINFOD_URLS@" ]; then
+	DEBUGINFOD_URLS="${DEBUGINFOD_URLS-}${DEBUGINFOD_URLS:+ }@DEBUGINFOD_URLS@"
+	export DEBUGINFOD_URLS
+fi
-- 
2.29.2

